### DISCUSSION TITLE
Nested url do problem...

### DISCUSSION THREAD

**Discussion by tomtastic** [Homebrew/discussions#3828] [Tap maintenance and Homebrew development]
I've been trying to write a formula for Wickr that requests the download URL via their API using a `POST` request :
```ruby
  url "https://enterprise-download.wickr.com/api/download",
    using: :post,
    data: {
            "name": "Wickr-Enterprise-Mac-Production",
          }
```
Which happily returns the location :
```
"hxxps://appcenter-filemanagement-distribabc123xyz.azureedge.net/a8cf3650-91e8-417f-b6e1-6daa5a676376/WickrEnterprise-5.112.14.dmg?sv=2019-02-02&sr=c&sig=%2B%2BFcmlgjqJehXFTLEGWcIbrAmR7khdoUwt2tqzRusjM%3D&se=2022-10-26T07%3A14%3A53Z&sp=r"
```
But I can't determine from any of the documentation how one uses nested-url-do's to take that URL and download it?

Would someone help me please?

**Comment by p-linnane** [comment]
This is not something we currently support.

**Reply by tomtastic** [reply]
So it's not possible to do in some manner of two-stage process?

**Reply by p-linnane** [reply]
Not currently, but PR's are always welcome.

**Reply by tomtastic** [reply]
OK, that wasn't very clear from the docs, so I've resorted to this :
![image](https://user-images.githubusercontent.com/302127/197984693-9e8aa2ad-7af9-42d5-be44-d1499a318fed.png)

**Reply by carlocab** [reply]
You can _possibly_ simplify that to something this: https://github.com/Homebrew/homebrew-cask-versions/blob/affd32c63a674c3a732ea7a2f7497718ece78029/Casks/transmission-nightly.rb#L5-L8

Note that this will cause issues for users if the base URL is unreachable. See https://github.com/Homebrew/homebrew-cask/issues/134356, https://github.com/Homebrew/brew/issues/14052.
